sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/CollapseProjectSuite.scala

```diff
@@ -223,6 +224,24 @@ class CollapseProjectSuite extends PlanTest {
     assert(hasMetadata(optimized))
   }
 
+  test("reset top-level alias metadata while collapsing projects") {
+    def hasNoMetadata(logicalPlan: LogicalPlan): Boolean = {
+      !logicalPlan.asInstanceOf[Project].projectList.exists(_.metadata.contains("key"))
+    }
+
+    val metadata = new MetadataBuilder().putLong("key", 1).build()
+    val analyzed =
+      Project(Seq(Alias('a_with_metadata, "b")(explicitMetadata = Some(Metadata.empty))),
+        Project(Seq(Alias('a, "a_with_metadata")(explicitMetadata = Some(metadata))),
+          testRelation.logicalPlan)).analyze
+    require(hasNoMetadata(analyzed))
```

Comment:
Although the upper `Alias` have an explicitly empty Metadata, it also have the metadata collect from child `Alias`.

Reply:
`Alias` could inherit metadata from child.